dali/kernels/imgproc/color_manipulation/brightness_contrast_test.cc

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <vector>
+#include <tuple>
+#include "dali/kernels/imgproc/color_manipulation/brightness_contrast.h"
+#include "dali/kernels/test/tensor_test_utils.h"
+
+namespace dali {
+namespace kernels {
+namespace test {
+
+namespace brightness_contrast {
+
+template<template<typename A, typename B> class Pair, typename T1, typename... T>
```

Comment:
This seems to be useful beyond this test - I assume we will have more cases where we need to test different combinations of input/output types. How about moving this to some test utils?

Reply:
I'm certain, that we will use it in the future and it should be moved to test utils, however, I'd do it in next possible moment, since then we will know more about how this "API" should look like. Now it's code written specifically for this particular case, next time there will be 2 distinct cases, so more knowledge and stuff